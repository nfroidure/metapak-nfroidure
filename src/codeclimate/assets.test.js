'use strict';

const fs = require('fs');
const path = require('path');
const assert = require('assert');
const assetsTransformer = require('./assets');

describe('TravisCI', () => {
  describe('Assets transformer', () => {
    it('should fill the manifest', () => {
      expect(
        assetsTransformer(
          {
            name: '.codeclimate.yml',
            data: fs.readFileSync(
              path.join(__dirname, 'assets', '_dot_codeclimate.yml'),
              'utf-8'
            ),
          },
          {
            metapak: {
              data: {
                files: 'src/*.js src/**/*.js',
                testsFiles: "test/*.js 'src/**/*.test.js'",
              },
            },
          }
        )
      ).toMatchSnapshot();
    });

    it('should let pass other files', () => {
      assert.deepEqual(
        assetsTransformer(
          {
            name: 'YOLO',
            data: 'Carpe diem\n',
          },
          {}
        ),
        {
          name: 'YOLO',
          data: 'Carpe diem\n',
        }
      );
    });
  });
});
