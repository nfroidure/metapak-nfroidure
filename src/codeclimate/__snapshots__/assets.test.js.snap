// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravisCI Assets transformer should fill the manifest 1`] = `
{
  "data": "# This file is automatically generated by a \`metapak\`
# module. Do not change it elsewhere, changes would
# be overridden.
engines:
  eslint:
    enabled: true

ratings:
   paths:
   - "src/*.js src/**/*.js"
## Exclude test files.
exclude_patterns:
- "dist/"
- "**/node_modules/"
- "test/*.js",
- "src/**/*.test.js"
",
  "name": ".codeclimate.yml",
}
`;
